### PR TITLE
Added more unit tests for legacy identity

### PR DESCRIPTION
**Pull request by liljenback**
More Identity testing and some name refactoring

## Related Issue

CORE-43419

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [x] I have run the Sandbox successfully.

### CODE REVIEW

**Comment by jfkhoury** [comment]
Hey @liljenback , can you please rebase on master, run npm ci locally and force push into this branch? This should fix the E2E tests.

File: src/components/Identity/createLegacyIdentity.js [hunk 1]
```diff
@@ -48,21 +51,21 @@ export default ({ idMigrationEnabled, orgId, logger }) => {
...
       if (idMigrationEnabled) {
         const ecid = getEcidFromLegacyCookies();
 
         if (ecid) {
           return Promise.resolve(ecid);
         }
 
-        if (doesVisitorExist) {
-          return getVisitorECID({ logger, orgId, awaitVisitorOptIn });
+        if (Visitor) {
```
Comment (jonsnyder) [inline src/components/Identity/createLegacyIdentity.js:62]: The only thing you are using Visitor for is to check if it exists.  If you change getEcidFromVisitor to return Promise.resolve() if Visitor is not there, then you don't need to have getVisitor as a dependency of createLegacyIdentity.js

File: test/unit/specs/components/Identity/visitorService/getVisitor.spec.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+describe("getVisitor", () => {
+  const mockWindow = {};
+  mockWindow.Visitor = jasmine.createSpy();
+  mockWindow.Visitor.getInstance = jasmine.createSpy();
+
+  it("Returns Visitor function if present and valid", () => {
+    expect(getVisitor(mockWindow)).toEqual(mockWindow.Visitor);
+  });
+});
```
[the remark this answers is not in the document]
Comment (liljenback) [reply test/unit/specs/components/Identity/visitorService/getVisitor.spec.js:23]: Sure

File: src/components/Identity/createLegacyIdentity.js [hunk 3]
```diff
@@ -48,21 +51,21 @@ export default ({ idMigrationEnabled, orgId, logger }) => {
...
+          return getEcidFromVisitor({ logger, orgId, awaitVisitorOptIn });
```
[the remark this answers is not in the document]
Comment (liljenback) [reply src/components/Identity/createLegacyIdentity.js:63]: Right, makes sense

File: src/components/Identity/createLegacyIdentity.js [hunk 4]
```diff
@@ -48,21 +44,17 @@ export default ({ idMigrationEnabled, orgId, logger }) => {
...
         const ecid = getEcidFromLegacyCookies();
-
         if (ecid) {
           return Promise.resolve(ecid);
         }
-
-        if (doesVisitorExist) {
-          return getVisitorECID({ logger, orgId, awaitVisitorOptIn });
-        }
+        return getEcidFromVisitor();
```
Comment (jonsnyder) [inline src/components/Identity/createLegacyIdentity.js:53]: 👍

File: test/unit/specs/components/Identity/visitorService/getVisitor.spec.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    mockWindow.Visitor.getInstance = jasmine.createSpy();
+    expect(getVisitor(mockWindow)).toEqual(mockWindow.Visitor);
+  });
+  it("Returns false if Visitor is available but does not support getInstance", () => {
+    mockWindow.Visitor = jasmine.createSpy();
+    expect(getVisitor(mockWindow)).toBe(false);
+  });
+  it("Returns false if Visitor is not available", () => {
+    expect(getVisitor(mockWindow)).toBe(false);
+  });
```
Comment (jonsnyder) [inline test/unit/specs/components/Identity/visitorService/getVisitor.spec.js:33]: 👍

**Pull request merged**